test/ttmlir/Dialect/TTIR/hoist/simple_add.mlir

```diff
@@ -0,0 +1,37 @@
+// RUN: ttmlir-opt --ttir-cpu-hoist-transform="hoist-locations=add_op1,add_op2,add_op3,add_op4" %s | tee /tmp/output.mlir | FileCheck %s
```

Comment:
@nsmithtt @vwellsTT would it make sense to instead of using loc we use optional attribute on ops (i.e should_hoist). To me it's more expressive and simplifies the check which ops need to be hoisted.

Reply:
Good idea, I like that a lot more.  Then I don't need specific param => I can have single analysis pass, vs "manual" and "automatic", etc.  I will sync w/ Nick on this